### ISSUE TITLE
ScrollTop scrolls itself / Restoring scroll position on Ons.Page for two Ons.List items

### ISSUE THREAD

**Issue by mittermayr** [OnsenUI/OnsenUI#2456]
__Environment__
Onsen UI Version:
2.10.1

Framework:
ReactJS

Framework binding:
React/Onsen

Platform:
Chrome, OSX, latest version (67) (viewing mobile/responsive view in Developer Console)
to be hosted inside a Cordova app (iOS, Android).

__Encountered problem__

Switching between two `Ons.List` will magically (wrongly) increase `scrollTop` of the `<Ons.Page>`, without the user (me) actually doing any scrolling in between switching. This does totally work fine and as expected when leaving the lists at scrollTop 0px. But once I scroll one list to 150px or so (and leave it there for good), then every switch back and forth, it will _magically_ adjust `scrollTop` upwards, further and further. The page scrolls, when it shouldn't.

_Here is the app layout_
I have a tab bar at the bottom app, and the content is a simple page, with a navigation bar and then a segment control on top. Below the segment control, there is a list. Switching the segment presents me with LIST A, or LIST B. Basically like this: `Ons.Page > Ons.Segment > Ons.List | Ons.List`

__How to reproduce__
I ran out of time trying to make a minimal reproducible example on the playground, but I'll try to get something together if this can't be solved otherwise. But this is pretty straight forward to understand, setup-wise:

Step 1:
I scroll _List A_, let's say to scrollTop 150px (needs to be more than 50 or so for this bug to happen). 

Step 2:
Now switch the segment to the second list, which now has the same scroll offset as _List A_, exactly at 150px. This is kind of expected, once you know that `Ons.Page` is the keeper of `scrollTop`. All good.

Step 3:
Now, I switch back to List A. The scrollTop has changed, and it's 175px now. I have not scrolled anything. None of my components do any scroll-fixing, nothing. The list has scrolled itself a tiny little bit.

Step 4:
If I keep switching back and forth, the list will scroll further and further, without me touching it.

Extra note: 
If I leave the lists at 0, and then switch back and forth, everything will stay at 0 as expected. Nothing scrolls down by itself.

**Comment by asialgearoid** [comment]
Thanks for the report. I tried to reproduce this without success. Here is a Codepen of what I tried. Please feel free to edit it to see if you can reproduce it there: https://codepen.io/anon/pen/GGwZOY

**Comment by mittermayr** [comment]
Terribly late with my response, but I finally had a chance to build a barebones example of what I have roughly going on with a bigger app. The issue can't be reproduced there, which is very odd.

So I guess you can close this issue now, and if I can manage to reproduce it in the barebones app, then I will ask to re-open. I am not sure where the app I am working on and the barebones version differ, to be honest, that part is almost 1:1, but I'll probably find it one day.

For anyone looking through the archives and hints on restoring scroll position, here's the barebones version for React, where a Segment switches between two lists and restores scroll positions on each. [Onsen List Scroll Restore Barebones Example](https://github.com/mittermayr/onsen-list-scroll-restore)

Thanks again for taking the time to writing the vanilla JS demo.

**Comment by mittermayr** [comment]
Another update: I have **found** the issue, finally. This is an issue to do with React-LazyLoad, which I had used as a replacement for the ons-lazy-repeat functionality, to have more fine-grained control over how the lazy-loading should happen. This worked perfectly in all cases, except for scroll restoration. React-LazyLoad requires to hook into the `onScroll` event of `Ons.Page` to function at all, and that, possibly in combination with the hard-coded (!) row height `LazyLoad height={...}`, this seems to cause trouble. 

Again, sorry about wasting anyone's time with this, I had removed LazyLoad from one of the two lists to make sure it wasn't interfering, but saw the same problem. Having left LazyLoad enabled on the second list was enough to still cause the issue on the first list. Reading a lot of 2017/2018 issues related to scrolling, I figured this may be another one coming to light. It was not.

Can be closed.
